lib/linguist/languages.yml

```diff
@@ -4305,6 +4305,24 @@ Ring:
   tm_scope: source.ring
   ace_mode: text
   language_id: 431
+Roblox:
+  type: data
+  tm_scope: none
+  ace_mode: text
+  extensions:
+  - ".rbxl"
+  - ".rbxm"
+Roblox XML:
```

Comment:
I don't think we should be calling out "Roblox XML" out as a different language as it is clearly plain old XML and sets an unnecessary precedent.

It is also a data language as you even acknowledge: "but the XML files contain important data for projects" 😄 

If you want to be able to identify your files as "Roblox", I suggest removing the "Roblox" language section (those files are binaries so not a programming language anyway) and rename "Roblox XML" to "Roblox", change the type to "data", and as @pchaigno suggested, identify these files as generated too.  Once Linguist knows of the language you can then implement an override as and when you like.

Reply:
Hi.  I haven't got back to you sooner.  I did the changes as you suggested (I'm still novice to git).  So here you go.